camille/processors/lidar.py

```diff
@@ -0,0 +1,114 @@
+from datetime import date
+from math import cos, sin, log, sqrt, radians
+import typing
+
+import pandas as pd
+import numpy as np
+from matplotlib import pyplot as plt
+
+Sample = typing.NewType('Sample', int)
+
+__defaults = {
+    'sample_dist': [50, 80, 120, 160, 200, 240, 280, 320, 360, 400],
+    'zeniths': list(map(radians, [15.74, 15.74, 15.74, 15.74])),
+    'azimuths': list(map(radians, [18.08, 161.92, -18.08, -161.92])),
+    'hub_hight': 98.6,
+    'lidar_height': 4.5,
+    'pitch_offset': radians(-2.0),
+    'roll_offset': radians(0.4),
+}
+
+
+def process(df, **kwargs):
+    options = ('sample_dist', 'zeniths', 'azimuths', 'hub_height',
+               'lidar_height', 'rotor_distance', 'pitch_offset', 'roll_offset')
+    if not all([key in options for key in kwargs.keys()]):
+        raise ArgumentError(
+            'Unknown argument(s) {}'.format(kwargs.keys() - options))
+
+    sample_dist = kwargs.get('sample_dist', __defaults['sample_dist'])
+    zeniths = kwargs.get('zeniths', __defaults['zeniths'])
+    azimuths = kwargs.get('azimuths', __defaults['azimuths'])
+    hub_hight = kwargs.get('hub_hight', __defaults['hub_hight'])
+    lidar_height = kwargs.get('lidar_height', __defaults['lidar_height'])
+    pitch_offset = kwargs.get('pitch_offset', __defaults['pitch_offset'])
+    roll_offset = kwargs.get('roll_offset', __defaults['roll_offset'])
+
+    sensors = list(range(4))
+
+    def group(df):
+        pattern = [0] * 10 + [1] * 10 + [2] * 10 + [3] * 10
+        L = []
+        i = 0
+        while i + 40 < len(df):
```

Comment:
As a first time reviewing i don't know the magic number 40 here. Guess it's obvious when you know the code, but - would it make sense making a variable with a describing name end replace all 40 's with this?